python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -112,6 +131,58 @@ def get_upstream_partitions_for_partition_range(
     ) -> PartitionKeyRange:
         raise NotImplementedError()
 
+    def downstream_partition_has_valid_upstream_partitions(
+        self,
+        downstream_partitions_def: PartitionsDefinition,
+        downstream_partition_key: str,
+        upstream_partitions_def: PartitionsDefinition,
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> bool:
+        if not isinstance(downstream_partitions_def, TimeWindowPartitionsDefinition):
```

Comment:
if all the callsites need to do this check, should we just put these checks inside _map_partitions?